pkg/api/types.go

```diff
@@ -121,3 +121,32 @@ type InstallResult struct {
 	// Error describes last error encountered during install operation
 	Error error
 }
+
+// SourceInfo stores information about the source code
+type SourceInfo struct {
+	// Ref represents a commit SHA-1, valid GIT branch name or a GIT tag
+	// The output image will contain this information as 'io.openshift.build.commit.ref' label.
+	Ref string
+
+	// CommitID represents an arbitrary extended object reference in GIT as SHA-1
+	// The output image will contain this information as 'io.openshift.build.commit.id' label.
+	CommitID string
+
+	// Date contains a date when the committer created the commit.
+	// The output image will contain this information as 'io.openshift.build.commit.date' label.
+	Date string
+
+	// Author contains information about the committer name and email address.
+	// The output image will contain this information as 'io.openshift.build.commit.author' label.
+	Author string
+
+	// SourceLocation contains a valid URL to the original repository.
+	// The output image will contain this information as 'io.openshift.build.source-location' label.
+	SourceLocation string
```

Comment:
If the struct is called `SourceInfo`, having this field named just `Location` would be enough, the same applies to `SourceContextDir` -> `ContextDir`.